### PR TITLE
Process media pushes

### PR DESCRIPTION
**Pull request by Pavel-Cherentsov**

### CODE REVIEW

File: app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt [hunk 1]
```diff
@@ -85,9 +87,69 @@ class FCMManagerImpl(
...
+                        notificationIcon = R.drawable.ic_media_image //R.drawable.ic_take_photo
+                    } else if (contentType.startsWith("video/")) {
+                        notificationIcon = R.drawable.ic_media_video
+                    } else if (contentType.startsWith("audio/")) {
+                        notificationIcon = R.drawable.ic_media_audio
+                    }
+                }
+            }
+
+            return createBitmapFromVectorDrawable(notificationIcon)
```
Comment (rusmonster) [inline app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt:122]:
```suggestion
            return ContextCompat.getDrawable(context, drawableId)?.toBitmap()
```

File: app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt [hunk 2]
```diff
@@ -85,9 +87,69 @@ class FCMManagerImpl(
...
+        }
+
+        val bitmap = Bitmap.createBitmap(200, 200, Bitmap.Config.ARGB_8888)
+        val canvas = Canvas(bitmap)
+        drawable.setBounds(0, 0, canvas.getWidth(), canvas.getHeight())
+        drawable.draw(canvas)
+        return bitmap
+    }
+
+    fun getLargeIconForNotification(payload: NotificationPayload): Bitmap? {
```
Comment (rusmonster) [inline app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt:103]: let's make it more kotlinish?
```
    val NotificationPayload.largeIconId: Int? get() = when (type) {
        NEW_MESSAGE -> when {

            mediaCount > 1 -> R.drawable.ic_media_multiple_attachments

            mediaCount > 0 -> with(mediaContentType) {
                when {
                    startsWith("image/") -> R.drawable.ic_media_image

                    startsWith("video/") -> R.drawable.ic_media_video

                    startsWith("audio/") -> R.drawable.ic_media_audio

                    else -> null
                }
            }

            else -> null
        }

        else -> null
    }

    val NotificationPayload.largeIcon get() = largeIconId?.let { ContextCompat.getDrawable(context, it) }
```

```
    payload.largeIcon?.let { notificationBuilder.setLargeIcon(it.toBitmap()) }
```

**Review by rusmonster — CHANGES_REQUESTED** [review]

**Comment by berkus** [comment]
Same review comments as @rusmonster left.

File: app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt [hunk 3]
```diff
@@ -85,9 +87,69 @@ class FCMManagerImpl(
...
+    }
 
+    fun getContentTextForNotification(payload: NotificationPayload): String {
+        var contentText = payload.body
+
+        if (payload.type == NotificationPayload.Type.NEW_MESSAGE) {
+            val mediaCount = payload.mediaCount
+            if (mediaCount > 0) {
+                if (mediaCount > 1) {
+                    contentText = "Media message with $mediaCount attachments"
```
[the remark this answers is not in the document]
Comment (Pavel-Cherentsov) [reply app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt]: fixed

File: app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt (hunk 2 above)
Comment (Pavel-Cherentsov) [reply app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt]: fixed

File: app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt (hunk 1 above)
Comment (rusmonster) [reply app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt]: double checked - `toBitmap()` works ;)

File: app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt [hunk 4]
```diff
@@ -85,9 +87,62 @@ class FCMManagerImpl(
...
+                }
+            }
+            else -> R.drawable.ic_media_document
+        }
+        else -> R.drawable.ic_media_document
+    }
+
+    val NotificationPayload.largeIcon get() = largeIconId?.let { createBitmapFromVectorDrawable(it) }
+
+    fun getContentTextForNotification(payload: NotificationPayload): String {
```
Comment (rusmonster) [inline app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt:121]: plz make it `NotificationPayload.textForNotification` - the same way as `NotificationPayload.largeIconId` above. Readability much better...

**Comment by rusmonster** [comment]
Tests have to be fixed...

File: app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt (hunk 1 above)
Comment (Pavel-Cherentsov) [reply app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt]: fixed

File: app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt (hunk 4 above)
Comment (Pavel-Cherentsov) [reply app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt]: fixed

File: app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt [hunk 5]
```diff
@@ -85,9 +86,35 @@ class FCMManagerImpl(
         }
     }
 
-    override fun showNotification(payload: NotificationPayload) {
-        Timber.d("showNotification: ${payload.conversationSid}")
+    val NotificationPayload.textForNotification: String get() = when (type) {
+        NotificationPayload.Type.NEW_MESSAGE -> when {
+            mediaCount > 1 -> context.getString(R.string.notification_media_message, mediaCount)
+            mediaCount > 0 -> context.getString(R.string.notification_media) + ": " +
+                        mediaFilename.ifEmpty { android.text.format.Formatter.formatShortFileSize(context, mediaSize) }
```
Comment (rusmonster) [inline app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt:93]: import android.text.format.Formatter

Comment (Pavel-Cherentsov) [reply app/src/main/java/com/twilio/conversations/app/manager/FCMManager.kt]: fixed

**Review by rusmonster — APPROVED** [review]

**Pull request merged**